### PR TITLE
Add PyTorch Profiler in Tensorboard loging

### PR DESCRIPTION
**Pull request by ggrzeczkowicz**
Add an option in training.tensorboard to use PyTorch Profiler.

PyTorch Profiler required torch>=1.8.1

### CODE REVIEW

**Review by nicolas-chaulet — CHANGES_REQUESTED** [review]
Looks good, thanks a lot for the contribution! Could you just run `poetry update` at the root of the repo to make sure that poetry lock file is correct as well?

File: docker/install_python.sh [hunk 1]
```diff
@@ -10,13 +10,13 @@ python3 -m pip install -U pip
 pip3 install setuptools>=41.0.0
 if [ $1 == "gpu" ]; then
     echo "Install GPU"
-    pip3 install torch==1.7.0 torchvision==0.8.1
+    pip3 install torch==1.8.1 torchvision==0.8.1
```
Comment (nicolas-chaulet) [inline docker/install_python.sh:13]: Torch vision is not the correct version, you can actually remove it entirely I think

File: docker/install_python.sh [hunk 2]
```diff
@@ -10,13 +10,13 @@ python3 -m pip install -U pip
...
     pip3 install MinkowskiEngine --install-option="--force_cuda" --install-option="--cuda_home=/usr/local/cuda"
     pip3 install git+https://github.com/mit-han-lab/torchsparse.git@v1.4.0 -v
     pip3 install pycuda
 else
     echo "Install CPU"
-    pip3 install torch==1.7.0+cpu torchvision==0.8.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
+    pip3 install torch==1.8.1+cpu torchvision==0.8.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (nicolas-chaulet) [inline docker/install_python.sh:19]: Same as above

**Comment by ggrzeczkowicz** [comment]
> Looks good, thanks a lot for the contribution! Could you just run `poetry update` at the root of the repo to make sure that poetry lock file is correct as well?

I did it and obtained no error.

```
(torch-points3d-OZjxPrEt-py3.8) points3d@d516bd59bdb3:~/torch-points3d$ poetry update
Updating dependencies
Resolving dependencies... (202.9s)

Writing lock file

Package operations: 7 installs, 59 updates, 1 removal

  • Removing async-generator (1.10)
  • Updating traitlets (5.0.5 -> 5.1.0)
  • Updating python-dateutil (2.8.1 -> 2.8.2)
  • Updating pyzmq (22.1.0 -> 22.2.1)
  • Updating decorator (4.4.2 -> 5.1.0)
  • Updating jupyter-client (6.2.0 -> 7.0.2)
  • Updating matplotlib-inline (0.1.2 -> 0.1.3)
  • Updating prompt-toolkit (3.0.19 -> 3.0.20)
  • Updating pygments (2.9.0 -> 2.10.0)
  • Updating bleach (3.3.0 -> 4.1.0)
  • Updating cffi (1.14.5 -> 1.14.6)
  • Updating debugpy (1.3.0 -> 1.4.3)
  • Updating idna (2.10 -> 3.2)
  • Updating ipython (7.25.0 -> 7.27.0)
  • Installing multidict (5.1.0)
  • Updating nbclient (0.5.3 -> 0.5.4)
  • Updating argon2-cffi (20.1.0 -> 21.1.0)
  • Installing async-timeout (3.0.1)
  • Installing charset-normalizer (2.0.4)
  • Updating ipykernel (6.0.1 -> 6.4.1)
  • Updating send2trash (1.7.1 -> 1.8.0)
  • Updating terminado (0.10.1 -> 0.12.1)
  • Updating typing-extensions (3.10.0.0 -> 3.10.0.2)
  • Installing yarl (1.6.3)
  • Installing aiohttp (3.7.4.post0)
  • Updating cryptography (3.4.7 -> 3.4.8)
  • Updating kiwisolver (1.3.1 -> 1.3.2)
  • Updating notebook (6.4.0 -> 6.4.3)
  • Updating pillow (8.3.0 -> 8.3.2)
  • Updating requests (2.25.1 -> 2.26.0)
  • Updating threadpoolctl (2.1.0 -> 2.2.0)
  • Installing backports.entry-points-selectable (1.1.0)
  • Updating google-auth (1.32.1 -> 1.35.0)
  • Updating jupyterlab-widgets (1.0.0 -> 1.0.1)
  • Updating matplotlib (3.4.2 -> 3.4.3)
  • Updating networkx (2.5.1 -> 2.6.3)
  • Updating param (1.10.1 -> 1.11.1)
  • Installing platformdirs (2.3.0)
  • Updating twisted (21.2.0 -> 21.7.0)
  • Updating websocket-client (1.1.0 -> 1.2.1)
  • Updating wslink (0.2.0 -> 1.0.7)
  • Updating astroid (2.6.2 -> 2.7.3)
  • Updating cfgv (3.3.0 -> 3.3.1)
  • Updating gitpython (3.1.18 -> 3.1.23)
  • Updating google-auth-oauthlib (0.4.4 -> 0.4.6)
  • Updating grpcio (1.38.1 -> 1.40.0)
  • Updating identify (2.2.10 -> 2.2.14)
  • Updating importlib-resources (5.2.0 -> 5.2.2)
  • Updating ipywidgets (7.6.3 -> 7.6.4)
  • Updating isort (5.9.1 -> 5.9.3)
  • Updating meshio (4.4.6 -> 5.0.0)
  • Updating pathspec (0.8.1 -> 0.9.0)
  • Updating rdflib (5.0.0 -> 6.0.0)
  • Updating regex (2021.7.1 -> 2021.8.28)
  • Updating sentry-sdk (1.1.0 -> 1.3.1)
  • Updating tqdm (4.61.1 -> 4.62.2)
  • Updating virtualenv (20.4.7 -> 20.7.2)
  • Updating watchdog (2.1.3 -> 2.1.5)
  • Updating codecov (2.1.11 -> 2.1.12)
  • Updating h5py (3.3.0 -> 3.4.0)
  • Updating hydra-core (1.0.6 -> 1.0.7)
  • Updating jupyterlab (1.2.20 -> 1.2.21)
  • Updating pylint (2.9.3 -> 2.10.2)
  • Updating tensorboard (2.5.0 -> 2.6.0)
  • Updating torch-scatter (2.0.7 -> 2.0.8)
  • Updating torch-sparse (0.6.10 -> 0.6.12)
  • Updating types-six (0.1.7 -> 0.1.9)
```

**Pull request closed without merging**